File: codes/0xD0_suepacm/0xD0_competition/sit/L2-2/L2-2.cpp
#include<bits/stdc++.h>
#define int long long
using namespace std;
const int MAXN = 1e6 + 7;
const int mod = 1e9 + 7;
int a[MAXN];
int b[MAXN];
struct Node
{
    int i;
    int j;
    int value;
    bool operator()(Node x, Node y) {
        return x.value > y.value;
    }
};

void solve() {
    signed n, m, k;
    cin >> n >> m >> k;
    for (int i = 0;i < n;++i)cin >> a[i];
    for (int i = 0;i < m;++i)cin >> b[i];
    sort(a, a + n);
    sort(b, b + m);
    priority_queue<Node, vector<Node>, Node> pq;
    for (int i = 0; i < n; i++)
        pq.push({ i,0,a[i] + b[0] });
    int sum = 0;
    for (signed i = 0;i < k;++i) {
        Node cur = pq.top();
        pq.pop();
        sum += cur.value % mod;
        sum %= mod;
        if (cur.j < m - 1)
            pq.push({ cur.i,cur.j + 1,a[cur.i] + b[cur.j + 1] });
    }
    cout << sum;
}

signed main() {
    ios::sync_with_stdio(false);
    cin.tie(0), cout.tie(0);
    int t;cin >> t;
    while (t--) {
        solve();
        cout << '\n';
    }
    return 0;
}

File: codes/0xC0_suepacm/0xD0_competition/sit/L2-2/L2-2.py
import queue
mod = int(1e9+7)
t = int(input())
for __ in range(t):
    n,m,k = [int(x) for x in input().split()]
    a = [int(x) for x in input().split()]
    b = [int(x) for x in input().split()]
    a.sort()
    b.sort()
    pq = queue.PriorityQueue()

    for i in range(n):
        pq.put((a[i]+b[0],(i,0)))

    sum = 0
    for _ in range(k):
        cur = pq.get()
        sum += cur[0] % mod
        sum %= mod
        if cur[1][1] < m-1:
            pq.put((a[cur[1][0]]+b[cur[1][1]+1],(cur[1][0],cur[1][1]+1)))

    print(sum)